README.md
# SympluR

[![DOI](https://zenodo.org/badge/122243873.svg)](https://zenodo.org/badge/latestdoi/122243873)
[![CRAN_Status_Badge](http://www.r-pkg.org/badges/version-last-release/SympluR)](https://cran.r-project.org/package=SympluR)
[![Build Status](https://travis-ci.org/symplur/SympluR.svg?branch=master)](https://travis-ci.org/symplur/SympluR)
[![License: MIT](https://img.shields.io/badge/License-MIT-blue.svg)](https://github.com/HealthCatalystSLC/healthcareai-r/blob/master/LICENSE)

**[SympluR](https://CRAN.R-project.org/package=SympluR)** is an [R](https://www.r-project.org) package for analyzing data from the Healthcare Social Graph via access to the [Symplur API](https://api.symplur.com/v1/docs).

#### #hcsmR

Take a look at the over [200 published journal articles](https://www.symplur.com/healthcare-social-media-research/) that have employed or referenced Symplur data in their research. #hcsmR is a collaboration between Symplur and Stanford Medicine X.

## Get started

Install R package from CRAN:

```
install.packages("SympluR")
library(SympluR)
```

For latest development version, install the package from GitHub:

```
library("devtools")
install_github('symplur/SympluR')
library(SympluR)
```

## Symplur API Credentials

To make us of this R package you need to have access to the Symplur API. The package comes with demo credentials that allow you to access the demo dataset `#LCSMDemoData`. This dataset is a 4-week snapshot of the conversations from [#LCSM](https://www.symplur.com/healthcare-hashtags/lcsm/) (Lung Cancer Social Media) from 08/16/2017 to 09/15/2017.

_You can get a quick look at the data now by opening the same demo dataset in [Symplur Signals](https://dashboard.symplur.com/hashtag/LCSMDemoData?start=1504249200&end=1505458799) with a free account._

To access any other datasets, please [contact Symplur](https://www.symplur.com/contact/) for further details.

The R package will prompt you for your credentials during the first API call in your R session. Paste in your credentials when asked, or if you want to try out the demo data then hit enter without entering anything.

## Documentation

Find the documentation in R for each function in this package. Example:

`?symplurTweetsSummary`

`?symplurTweetsActivity`

`?symplurPeopleInfluencers`

`?symplurContentRetweets`

`?symplurContentWords`

To learn more about each Symplur API endpoint used in this package and the accepted parameters please see the [Symplur API Documentation](https://api.symplur.com/v1/docs).

## Example Usage

### Summary

The `symplurTweetsSummary()` function will create a list with statistics of the database and the time span selected. The stats includes tweets, mentions, impressions, users, retweets, replies, urls_shared, etc.

`LCSM <- symplurTweetsSummary("09/01/2017", "09/08/2017", databases = "#LCSMDemoData")`

### Summary Table

The `symplurTweetsSummaryTable()` function will create a data frame with summary statistics of all databases and time spans defined in an existing data frame.
First create in a spreadsheet columns 'database', 'start' and 'end'. Add rows according to your query needs, then export as a CSV-file. See [example CSV-file](https://github.com/symplur/SympluR/blob/master/inst/extdata/datasets.csv).

_Example table:_

| database      | start      | end        |
| ------------- | ---------- | ---------- |
| #LCSMDemoData | 09/01/2017 | 09/06/2017 |
| #LCSMDemoData | 09/06/2017 | 09/13/2017 |
| #LCSMDemoData | 09/13/2017 | 09/19/2017 |

Load such an CSV-file into R as a data frame:

```
library(readr)
LCSMquery <- read_csv(file.choose())
```

Now we're ready to run the analysis:

`LCSManalysis <- symplurTweetsSummaryTable(LCSMquery)`

You can also try out `symplurTweetsSummaryTable()` with an example CSV file:

```
library(readr)
datasets <- read_csv(system.file("extdata", "datasets.csv", package = "SympluR", mustWork = TRUE))
LCSMDemoDataTweetsSummaryTable <- symplurTweetsSummaryTable(datasets)
```

## Credits

Thank you to [Professor Larry Chu, MD](https://twitter.com/larrychu) at [Stanford University School of Medicine](https://medicinex.stanford.edu) for the idea of the SympluR R package.


NEWS.md
# SympluR 0.3.0

*   Supports access to the `twitter/analytics/people/influencers` endpoint of the Symplur API.

# SympluR 0.2.0

*   Supports access to the `twitter/analytics/tweets/activity` endpoint of the Symplur API.
*   Supports access to the `twitter/analytics/content/retweets` endpoint of the Symplur API.
*   Supports access to the `twitter/analytics/content/words` endpoint of the Symplur API.

# SympluR 0.1.0

*   First release.
*   Supports access to the `twitter/analytics/tweets/summary` endpoint of the Symplur API.
*   Provides a process for looping through multiple databases to create a large data frame.


R/symplurToken.R
#' Symplur API - Token
#'
#' @keywords token
#' @export
#' @examples
#' symplurToken()
#'
symplurToken <- function(){
  Sys.setenv(TZ="UTC")
  token <- Sys.getenv('SYMPLUR_TOKEN')
  config <- symplurConfig()

  if (identical(token, "")) {
    if (identical(Sys.getenv('SYMPLUR_clientId'), "")) {
      SymplurClientId <- readline("What is your Symplur API clientId? If none, just hit enter to try #LCSMDemoData: ")
      if (identical(SymplurClientId, "")) {
          SymplurClientId = config$demoClientId
      }
      Sys.setenv(SYMPLUR_clientId = SymplurClientId)
    }
    if (identical(Sys.getenv('SYMPLUR_clientSecret'), "")) {
      SymplurClientSecret <- readline("What is your Symplur API clientSecret? If none, just hit enter to try #LCSMDemoData: ")
      if (identical(SymplurClientSecret, "")) {
          SymplurClientSecret = config$demoClientSecret
      }
      Sys.setenv(SYMPLUR_clientSecret = SymplurClientSecret)
    }
    baseUrl = config$baseUrl
    clientId = Sys.getenv('SYMPLUR_clientId')
    clientSecret = Sys.getenv('SYMPLUR_clientSecret')
    endpoint = paste(baseUrl, "oauth/token", sep="")
    secret <- jsonlite::base64_enc(paste(clientId, clientSecret, sep = ":"))
    req <- httr::POST(endpoint,
      httr::add_headers(
      "Authorization" = paste("Basic", gsub("\n", "", secret)),
      "Content-Type" = "application/x-www-form-urlencoded;charset=UTF-8"
      ),
      body = "grant_type=client_credentials"
    );
    httr::stop_for_status(req, "Authenticate with Symplur")
    token <- httr::content(req)$access_token
    Sys.setenv(SYMPLUR_TOKEN = token)
  }
  return(token)
}


R/symplurSummary.R
#' Symplur API - Summary
#'
#' API endpoint documentation: https://api.symplur.com/v1/docs#/Twitter_Analytics:_Tweets/get_twitter_analytics_tweets_summary
#' @param start Start time for period analyzed, read above API docs for more info.
#' @param end End time for period analyzed
#' @param databases The database(s) analyzed. Comma separate string if using more than one database.
#' @keywords summary
#' @export
#' @examples
#' LCSMDemoData <- symplurSummary("09/01/2017", "09/08/2017", databases = "#LCSMDemoData")
#' @import httr
#' @import jsonlite
#'
symplurSummary <- function(start = "09/01/2017", end = "09/08/2017", databases = "#LCSMDemoData"){
  endpoint = "twitter/analytics/tweets/summary"
  parameters = list(
    "start" = start,
    "end" = end,
    "databases" = databases
    )

  config <- symplurConfig()
  url <- paste(config$baseUrl, endpoint, sep="")
  req <- httr::GET(url, query = parameters, httr::add_headers(Authorization = paste("Bearer", symplurToken())))
  json <- httr::content(req, as = "text", encoding = "UTF-8")
  summary <- fromJSON(json, flatten=TRUE)

  return(summary)
}



#' Symplur API - Summary Table
#'
#' Creates a dataframe from looping through queries.
#'
#' Example query dataframe:
#' \tabular{lll}{
#' database \tab start \tab end\cr
#' #BCSM \tab 01/01/2010 \tab 01/01/2018\cr
#' #LCSM \tab 01/01/2010 \tab 01/01/2018\cr
#' #BTSM \tab 01/01/2010 \tab 01/01/2018
#' }
#' @param query A dataframe with columns: database, start and end.
#' @keywords summary table
#' @export
#' @import plyr
#'
symplurSummaryTable <- function(query = data.frame(database=character(),start=character(), end=character())){
  for (i in 1:nrow(query)) {
    database = as.character(query[i, "database"])
    start = as.character(query[i, "start"])
    end = as.character(query[i, "end"])

    message("Now analyzing: ", database)
    start.time <- Sys.time()
    data <- symplurSummary(start,end,database)
    end.time <- Sys.time()
    time.taken <- end.time - start.time
    message("Execution time: ", time.taken)

    if (i == 1){
      table <- as.data.frame(data)
    } else {
      table <- rbind.fill(table, as.data.frame(data))
    }
  }
  return(table)
}


R/symplurContentWords.R
#' Symplur API - Content/Words
#'
#' API endpoint documentation:
#' https://api.symplur.com/v1/docs#/Twitter_Analytics:_Content/get_twitter_analytics_content_words
#' @param start Start time for period analyzed, read above API docs for more info.
#' @param end End time for period analyzed
#' @param databases The database(s) analyzed. Comma separate string if using more than one database.
#' @param limit Maximum number of words to return. Integer.
#' @keywords content words
#' @export
#' @examples
#' LCSMDemoDataContentWords <- symplurContentWords(
#'     "09/01/2017",
#'     "09/08/2017",
#'     databases = "#LCSMDemoData",
#'     50)
#' @import httr
#' @import jsonlite
#'
symplurContentWords <- function(start = "09/01/2017", end = "09/08/2017", databases = "#LCSMDemoData", limit = 100){
  endpoint = "twitter/analytics/content/words"
  parameters = list(
    "start" = start,
    "end" = end,
    "databases" = databases,
    "limit" = limit
    )

  config <- symplurConfig()
  url <- paste(config$baseUrl, endpoint, sep="")
  req <- httr::GET(url, query = parameters, httr::add_headers(Authorization = paste("Bearer", symplurToken())))
  json <- httr::content(req, as = "text", encoding = "UTF-8")

  output <- fromJSON(json, flatten=TRUE)
  table <- as.data.frame(output)

  return(table)
}



#' Symplur API - Content/Words Table
#'
#' Creates a dataframe from looping through queries.
#'
#' Example query dataframe:
#' \tabular{lll}{
#' database \tab start \tab end\cr
#' #BCSM \tab 01/01/2010 \tab 01/01/2018\cr
#' #LCSM \tab 01/01/2010 \tab 01/01/2018\cr
#' #BTSM \tab 01/01/2010 \tab 01/01/2018
#' }
#' @param query A dataframe with columns: database, start and end.
#' @keywords content words table
#' @export
#' @examples
#' require(readr)
#' datasets <- read_csv(system.file("extdata", "datasets.csv", package = "SympluR", mustWork = TRUE))
#' LCSMDemoDataContentWordsTable <- symplurContentWordsTable(datasets)
#' @import plyr
#'
symplurContentWordsTable <- function(query = data.frame(database=character(),start=character(), end=character())){
  for (i in 1:nrow(query)) {
    database = as.character(query[i, "database"])
    start = as.character(query[i, "start"])
    end = as.character(query[i, "end"])

    message("Now analyzing: ", database)
    start.time <- Sys.time()
    data <- symplurContentWords(start,end,database)
    end.time <- Sys.time()
    time.taken <- end.time - start.time
    message("Execution time: ", time.taken)

    if (i == 1){
      table <- as.data.frame(data)
    } else {
      table <- rbind.fill(table, as.data.frame(data))
    }
  }
  return(table)
}


R/symplurTweetsSummary.R
#' Symplur API - Tweets/Summary
#'
#' API endpoint documentation:
#' https://api.symplur.com/v1/docs#/Twitter_Analytics:_Tweets/get_twitter_analytics_tweets_summary
#' @param start Start time for period analyzed, read above API docs for more info.
#' @param end End time for period analyzed
#' @param databases The database(s) analyzed. Comma separate string if using more than one database.
#' @keywords tweets summary
#' @export
#' @examples
#' LCSMDemoDataTweetsSummary <- symplurTweetsSummary(
#'    "09/01/2017",
#'    "09/08/2017",
#'    databases = "#LCSMDemoData")
#' @import httr
#' @import jsonlite
#'
symplurTweetsSummary <- function(start = "09/01/2017", end = "09/08/2017", databases = "#LCSMDemoData"){
  endpoint = "twitter/analytics/tweets/summary"
  parameters = list(
    "start" = start,
    "end" = end,
    "databases" = databases
    )

  config <- symplurConfig()
  url <- paste(config$baseUrl, endpoint, sep="")
  req <- httr::GET(url, query = parameters, httr::add_headers(Authorization = paste("Bearer", symplurToken())))
  json <- httr::content(req, as = "text", encoding = "UTF-8")

  output <- fromJSON(json, flatten=TRUE)
  table <- as.data.frame(output)

  return(table)
}



#' Symplur API - Tweets/Summary Table
#'
#' Creates a dataframe from looping through queries.
#'
#' Example query dataframe:
#' \tabular{lll}{
#' database \tab start \tab end\cr
#' #BCSM \tab 01/01/2010 \tab 01/01/2018\cr
#' #LCSM \tab 01/01/2010 \tab 01/01/2018\cr
#' #BTSM \tab 01/01/2010 \tab 01/01/2018
#' }
#' @param query A dataframe with columns: database, start and end.
#' @keywords tweets summary table
#' @export
#' @examples
#' require(readr)
#' datasets <- read_csv(system.file("extdata", "datasets.csv", package = "SympluR", mustWork = TRUE))
#' LCSMDemoDataTweetsSummaryTable <- symplurTweetsSummaryTable(datasets)
#' @import plyr
#'
symplurTweetsSummaryTable <- function(query = data.frame(database=character(),start=character(), end=character())){
  for (i in 1:nrow(query)) {
    database = as.character(query[i, "database"])
    start = as.character(query[i, "start"])
    end = as.character(query[i, "end"])

    message("Now analyzing: ", database)
    start.time <- Sys.time()
    data <- symplurTweetsSummary(start,end,database)
    end.time <- Sys.time()
    time.taken <- end.time - start.time
    message("Execution time: ", time.taken)

    if (i == 1){
      table <- as.data.frame(data)
    } else {
      table <- rbind.fill(table, as.data.frame(data))
    }
  }
  return(table)
}


R/symplurTweetsActivity.R
#' Symplur API - Tweets/Activity
#'
#' API endpoint documentation:
#' https://api.symplur.com/v1/docs#/Twitter_Analytics:_Tweets/get_twitter_analytics_tweets_activity
#' @param start Start time for period analyzed, read above API docs for more info.
#' @param end End time for period analyzed
#' @param databases The database(s) analyzed. Comma separate string if using more than one database.
#' @param interval Unit of time to use for periods. Supported values are "minutes", "hours", "days", "weeks", "months", and "years". You must choose an interval that results in 500 periods or less, based on your selected date range. If omitted, the system will try to guess an appropriate interval.
#' @keywords tweets activity
#' @export
#' @examples
#' LCSMDemoDataTweetsActivity <- symplurTweetsActivity(
#'     "09/01/2017",
#'     "09/08/2017",
#'     databases = "#LCSMDemoData",
#'     "days")
#' @import httr
#' @import jsonlite
#'
symplurTweetsActivity <- function(start = "09/01/2017", end = "09/08/2017", databases = "#LCSMDemoData", interval = "days"){
  endpoint = "twitter/analytics/tweets/activity"
  parameters = list(
    "start" = start,
    "end" = end,
    "databases" = databases,
    "interval" = interval
    )

  config <- symplurConfig()
  url <- paste(config$baseUrl, endpoint, sep="")
  req <- httr::GET(url, query = parameters, httr::add_headers(Authorization = paste("Bearer", symplurToken())))
  json <- httr::content(req, as = "text", encoding = "UTF-8")

  options(warn=-1)
  output <- fromJSON(json, flatten=TRUE)
  table <- as.data.frame(output)
  table <- cbind("tweet count" = 0, table, row.names = NULL)
  table["tweet count"] <- output$series$data
  drops <- c("series.name", "series.data", "request.start", "request.end")
  table <-  table[ , !(names(table) %in% drops)]
  options(warn=0)

  return(table)
}



#' Symplur API - Tweets/Activity Table
#'
#' Creates a dataframe from looping through queries.
#'
#' Example query dataframe:
#' \tabular{lll}{
#' database \tab start \tab end\cr
#' #BCSM \tab 01/01/2010 \tab 01/01/2018\cr
#' #LCSM \tab 01/01/2010 \tab 01/01/2018\cr
#' #BTSM \tab 01/01/2010 \tab 01/01/2018
#' }
#' @param query A dataframe with columns: database, start and end.
#' @keywords tweets activity table
#' @export
#' @examples
#' require(readr)
#' datasets <- read_csv(system.file("extdata", "datasets.csv", package = "SympluR", mustWork = TRUE))
#' LCSMDemoDataTweetsActivityTable <- symplurTweetsActivityTable(datasets)
#' @import plyr
#'
symplurTweetsActivityTable <- function(query = data.frame(database=character(),start=character(), end=character())){
  for (i in 1:nrow(query)) {
    database = as.character(query[i, "database"])
    start = as.character(query[i, "start"])
    end = as.character(query[i, "end"])

    message("Now analyzing: ", database)
    start.time <- Sys.time()
    data <- symplurTweetsActivity(start,end,database)
    end.time <- Sys.time()
    time.taken <- end.time - start.time
    message("Execution time: ", time.taken)

    if (i == 1){
      table <- as.data.frame(data)
    } else {
      table <- rbind.fill(table, as.data.frame(data))
    }
  }
  return(table)
}


R/symplurContentRetweets.R
#' Symplur API - Content/Retweets
#'
#' API endpoint documentation:
#' https://api.symplur.com/v1/docs#/Twitter_Analytics:_Content/get_twitter_analytics_content_retweets
#' @param start Start time for period analyzed, read above API docs for more info.
#' @param end End time for period analyzed
#' @param databases The database(s) analyzed. Comma separate string if using more than one database.
#' @param limit Maximum number of retweets to return. Must be positive integer. Default is 10. Max is 500.
#' @keywords content retweets
#' @export
#' @examples
#' LCSMDemoDataContentRetweets <- symplurContentRetweets(
#'     "09/01/2017",
#'     "09/08/2017",
#'     databases = "#LCSMDemoData",
#'     50)
#' @import httr
#' @import jsonlite
#'
symplurContentRetweets <- function(start = "09/01/2017", end = "09/08/2017", databases = "#LCSMDemoData", limit = 100){
  endpoint = "twitter/analytics/content/retweets"
  parameters = list(
    "start" = start,
    "end" = end,
    "databases" = databases,
    "limit" = limit
    )

  config <- symplurConfig()
  url <- paste(config$baseUrl, endpoint, sep="")
  req <- httr::GET(url, query = parameters, httr::add_headers(Authorization = paste("Bearer", symplurToken())))
  json <- httr::content(req, as = "text", encoding = "UTF-8")

  output <- fromJSON(json, flatten=TRUE)
  table <- as.data.frame(output)
  drops <- c("tweets.tweet.media")
  table <-  table[ , !(names(table) %in% drops)]
  for (i in 1:nrow(table)) {
      if (is.null(table$tweets.tweet.user.stakeholder_categories[i][[1]][["id"]])) {
          table$tweets.tweet.user.stakeholder_categories[i] = "NA"
      } else {
          table$tweets.tweet.user.stakeholder_categories[i] = paste(sapply(table$tweets.tweet.user.stakeholder_categories[i], `[[`, 2), collapse=', ')
      }
  }

  table$tweets.tweet.user.stakeholder_categories <- vapply(table$tweets.tweet.user.stakeholder_categories, paste, collapse = ", ", character(1L))

  return(table)
}



#' Symplur API - Content/Retweets Table
#'
#' Creates a dataframe from looping through queries.
#'
#' Example query dataframe:
#' \tabular{lll}{
#' database \tab start \tab end\cr
#' #BCSM \tab 01/01/2010 \tab 01/01/2018\cr
#' #LCSM \tab 01/01/2010 \tab 01/01/2018\cr
#' #BTSM \tab 01/01/2010 \tab 01/01/2018
#' }
#' @param query A dataframe with columns: database, start and end.
#' @keywords content retweets table
#' @export
#' @examples
#' require(readr)
#' datasets <- read_csv(system.file("extdata", "datasets.csv", package = "SympluR", mustWork = TRUE))
#' LCSMDemoDataContentRetweetsTable <- symplurContentRetweetsTable(datasets)
#' @import plyr
#'
symplurContentRetweetsTable <- function(query = data.frame(database=character(),start=character(), end=character())){
  for (i in 1:nrow(query)) {
    database = as.character(query[i, "database"])
    start = as.character(query[i, "start"])
    end = as.character(query[i, "end"])

    message("Now analyzing: ", database)
    start.time <- Sys.time()
    data <- symplurContentRetweets(start,end,database)
    end.time <- Sys.time()
    time.taken <- end.time - start.time
    message("Execution time: ", time.taken)

    if (i == 1){
      table <- as.data.frame(data)
    } else {
      table <- rbind.fill(table, as.data.frame(data))
    }
  }
  return(table)
}


R/symplurPeopleInfluencers.R
#' Symplur API - People/Influencers
#'
#' API endpoint documentation:
#' https://api.symplur.com/v1/docs#/Twitter_Analytics:_People/get_twitter_analytics_people_influencers
#' @param start Start time for period analyzed, read above API docs for more info.
#' @param end End time for period analyzed
#' @param databases The database(s) analyzed. Comma separate string if using more than one database.
#' @param limit Maximum number of retweets to return. Must be positive integer. Default is 10. Max is 500.
#' @param metric Comma-separated list of one or more metrics to calculate. Results will be sorted by the first metric in the list. Supported metrics are "hsg_score", "mentions", "retweets", "tweets", "impressions", and "replies".
#' @keywords people influencers
#' @export
#' @examples
#' LCSMDemoDataPeopleInfluencers <- symplurPeopleInfluencers(
#'     "09/01/2017",
#'     "09/08/2017",
#'     databases = "#LCSMDemoData",
#'     50)
#' @import httr
#' @import jsonlite
#'
symplurPeopleInfluencers <- function(start = "09/01/2017", end = "09/08/2017", databases = "#LCSMDemoData", limit = 100, metric = "mentions"){
  endpoint = "twitter/analytics/people/influencers"
  parameters = list(
    "start" = start,
    "end" = end,
    "databases" = databases,
    "limit" = limit,
    "metric" = metric
    )

  config <- symplurConfig()
  url <- paste(config$baseUrl, endpoint, sep="")
  req <- httr::GET(url, query = parameters, httr::add_headers(Authorization = paste("Bearer", symplurToken())))
  json <- httr::content(req, as = "text", encoding = "UTF-8")

  output <- fromJSON(json, flatten=TRUE)
  table <- as.data.frame(output)
  table <- cbind(influencers.rank = "", table)
  for (i in 1:nrow(table)) {
      if (is.null(table$influencers.user.stakeholder_categories[i][[1]][["id"]])) {
          table$influencers.user.stakeholder_categories[i] = "NA"
      } else {
          table$influencers.user.stakeholder_categories[i] = paste(sapply(table$influencers.user.stakeholder_categories[i], `[[`, 2), collapse=', ')
      }
  }

  table$influencers.user.stakeholder_categories <- vapply(table$influencers.user.stakeholder_categories, paste, collapse = ", ", character(1L))
  table$influencers.rank <- 1:nrow(table)
  return(table)
}



#' Symplur API - People/Influencers Table
#'
#' Creates a dataframe from looping through queries.
#'
#' Example query dataframe:
#' \tabular{lll}{
#' database \tab start \tab end\cr
#' #BCSM \tab 01/01/2010 \tab 01/01/2018\cr
#' #LCSM \tab 01/01/2010 \tab 01/01/2018\cr
#' #BTSM \tab 01/01/2010 \tab 01/01/2018
#' }
#' @param query A dataframe with columns: database, start and end.
#' @keywords people influencers table
#' @export
#' @examples
#' require(readr)
#' datasets <- read_csv(system.file("extdata", "datasets.csv", package = "SympluR", mustWork = TRUE))
#' LCSMDemoDataPeopleInfluencersTable <- symplurPeopleInfluencersTable(datasets)
#' @import plyr
#'
symplurPeopleInfluencersTable <- function(query = data.frame(database=character(),start=character(), end=character())){
  for (i in 1:nrow(query)) {
    database = as.character(query[i, "database"])
    start = as.character(query[i, "start"])
    end = as.character(query[i, "end"])

    message("Now analyzing: ", database)
    start.time <- Sys.time()
    data <- symplurPeopleInfluencers(start,end,database)
    end.time <- Sys.time()
    time.taken <- end.time - start.time
    message("Execution time: ", time.taken)

    if (i == 1){
      table <- as.data.frame(data)
    } else {
      table <- rbind.fill(table, as.data.frame(data))
    }
  }
  return(table)
}
